llvm/include/llvm/ADT/APInt.h

```diff
@@ -1688,6 +1696,16 @@ class [[nodiscard]] APInt {
     return APInt(sizeof(float) * CHAR_BIT, llvm::bit_cast<uint32_t>(V));
   }
 
+#ifdef __FLOAT128__
+  static APInt longDoubleToBits(float128 V) {
```

Comment:
I don't think APInt/APFloat should be exposing APIs that are only conditionally available. That kind of defeats the point of using them

Reply:
I agree, but I can't think of a better way to do this currently since logf128 isn't available on all targets but is the only type that guarantees a floating-point type of 128 bits.

Reply:
The libc logf128() function takes a float128 argument, and returns a float128 result.  If we're using it, we need to convert between float128 and APFloat.  (If APFloat::log() existed, this patch would be much simpler... but it doesn't, and it's difficult to write.)

Reply:
I think it's correct to remove this (and have done so) since APFloat already has internal representation of float128's. I'm not sure about the `convertToQuad` function I've added though, since that is necessary to extract the float128 information from APFloat in order to pass it to the C logf128 function.